processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -1004,4 +1029,28 @@ public boolean isLiteral() {
         return isLiteral;
     }
 
+    /**
+     * It strips the all the {@code []} from the {@code className}.
+     *
+     * E.g.
+     * <pre>
+     *     trimSimpleClassName("String[][][]") -> "String"
+     *     trimSimpleClassName("String[]") -> "String"
+     * </pre>
+     *
+     * @param className that needs to be trimmed
+     *
+     * @return the trimmed {@code className}, or {@code null} if the {@code className} was {@code null}
+     */
+    private String trimSimpleClassName(String className) {
```

Comment:
Maybe move this to a utility class as it might be useful also somewhere else in the future?

Reply:
Not sure... lets do that when the need arises..